File: dotnet/Middleware.Contracts/Events/AppraisalCompletedEvent.cs
namespace Middleware.Contracts.Events;

// UC4 · Appraisal Documents — StatusCode=15 workflow completed; appraisal closed
public sealed class AppraisalCompletedEvent
{
    public string AppraisalId { get; set; } = string.Empty;
    public string PolicyNumber { get; set; } = string.Empty;
    public string InspectionId { get; set; } = string.Empty;

    // ⚠️ DEMO GAP: Masterpiece Transaction 90 response schema unknown — REPLACE_ME_MASTERPIECE_TX90_RESPONSE
    public string MasterpieceTransactionRef { get; set; } = string.Empty;

    public string CorrelationId { get; set; } = string.Empty;
    public DateTimeOffset CompletedAt { get; set; }
}

File: java/common/src/main/java/com/ais/middleware/common/events/prs/AppraisalCompletedEvent.java
package com.ais.middleware.common.events.prs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published when StatusCode=15 (Completed) flow finishes.
 * Masterpiece Transaction 90 has been called and appraisal closed.
 *
 * Topic: prs.events.appraisal-completed
 */
public record AppraisalCompletedEvent(
        String correlationId,
        String appraisalId,
        String policyNumber,
        OffsetDateTime completedAt
) {
    @JsonCreator
    public AppraisalCompletedEvent(
            @JsonProperty("correlationId") String correlationId,
            @JsonProperty("appraisalId") String appraisalId,
            @JsonProperty("policyNumber") String policyNumber,
            @JsonProperty("completedAt") OffsetDateTime completedAt
    ) {
        this.correlationId = correlationId;
        this.appraisalId = appraisalId;
        this.policyNumber = policyNumber;
        this.completedAt = completedAt;
    }
}
